pkgs/applications/version-management/fossil/default.nix

```diff
@@ -23,6 +24,8 @@ stdenv.mkDerivation rec {
     sha256 = "1knff50rr8f39myxj50fprb9ya87cslmwz7zzfya56l33r7i7jh3";
   };
 
+  depsBuildBuild = [ buildPackages.stdenv.cc ];
```

Comment:
```suggestion
  # required for build time tool `./tools/translate.c`
  depsBuildBuild = [ buildPackages.stdenv.cc ];
```

Reply:
Should we have this in cross by default?

Reply:
no, this adds a `cc`/`gcc` command and so it would silently break cross-compilation due to possibly having wrong arch binaries in outputs

Reply:
Ah, I see. Yeah, some build system would such enough to fall into this.

Perhaps a simple toggle like `buildsInternalTools = true;` could be added to avoid having to depend on `buildPackages` in every derivation where this is the case.

Reply:
mkDerivation can't depend on pkgs and even if it could i wouldn't agree


if stdenv.cc was spliced then we wouldn't have to depend on buildPackages explicitly
(currently no `pkgsCross.aarch64-multiplatform.__splicedPackages.stdenv.cc.__spliced` )
related issue https://github.com/NixOS/nixpkgs/issues/211340

Reply:
> mkDerivation can't depend on pkgs and even if it could i wouldn't agree

Absolutely not, I agree. What I meant though is to have an optional build -> build stdenv in the cross stdenv to build tools used during the build.

Reply:
i think its better to keep it like it currently is as we already have it documented https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/cross-compilation.chapter.md#what-if-my-packages-build-system-needs-to-build-a-c-program-to-be-run-under-the-build-environment-cross-qa-build-c-program-in-build-environment